Finchley.SR2 keystore Startup errors

I upgraded the version of spring cloud from finchley.sr1 to finchley.sr2, and the version of spring boot was upgraded from 2.0.4.release to 2.0.6.release. When starting the config project, I got the following error:

```
java.lang.IllegalStateException: Cannot decrypt: key=spring.security.user.password
	at org.springframework.cloud.bootstrap.encrypt.EnvironmentDecryptApplicationInitializer.decrypt(EnvironmentDecryptApplicationInitializer.java:202)
	at org.springframework.cloud.bootstrap.encrypt.EnvironmentDecryptApplicationInitializer.decrypt(EnvironmentDecryptApplicationInitializer.java:166)
	at org.springframework.cloud.bootstrap.encrypt.EnvironmentDecryptApplicationInitializer.initialize(EnvironmentDecryptApplicationInitializer.java:96)
	at org.springframework.cloud.bootstrap.BootstrapApplicationListener$DelegatingEnvironmentDecryptApplicationInitializer.initialize(BootstrapApplicationListener.java:424)
	at org.springframework.boot.SpringApplication.applyInitializers(SpringApplication.java:628)
	at org.springframework.boot.SpringApplication.prepareContext(SpringApplication.java:364)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:305)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1242)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1230)
	at com.mchain.module.config.ConfigServerApplication.main(ConfigServerApplication.java:14)
Caused by: java.lang.IllegalStateException: Cannot load keys from store: class path resource [config-server.keystore]
	at org.springframework.security.rsa.crypto.KeyStoreKeyFactory.getKeyPair(KeyStoreKeyFactory.java:76)
	at org.springframework.cloud.config.server.encryption.KeyStoreTextEncryptorLocator.locate(KeyStoreTextEncryptorLocator.java:84)
	at org.springframework.cloud.config.server.encryption.LocatorTextEncryptor.decrypt(LocatorTextEncryptor.java:52)
	at org.springframework.cloud.bootstrap.encrypt.EnvironmentDecryptApplicationInitializer.decrypt(EnvironmentDecryptApplicationInitializer.java:194)
	... 9 common frames omitted
Caused by: java.security.KeyStoreException: keystore not found
	at java.security.KeyStore.getInstance(KeyStore.java:851)
	at org.springframework.security.rsa.crypto.KeyStoreKeyFactory.getKeyPair(KeyStoreKeyFactory.java:64)
	... 12 common frames omitted
Caused by: java.security.NoSuchAlgorithmException: keystore KeyStore not available
	at sun.security.jca.GetInstance.getInstance(GetInstance.java:159)
	at java.security.Security.getImpl(Security.java:695)
	at java.security.KeyStore.getInstance(KeyStore.java:848)
	... 13 common frames omitted
```
The error showed that the keystore file could not be found, but it was in our project path and the bootstrap.yml configuration was as follows:
```
encrypt:
  key-store:
    location: classpath:/config-server.keystore
    alias: config-server
    password: test
    secret: test
```
After I reset spring cloud and spring boot versions back to previous versions, the project was running normally, and I compared the finchley.sr1 and finchley.sr2 keystore documents without seeing any changes.

--- a/docs/src/main/asciidoc/spring-cloud-config.adoc
+++ b/docs/src/main/asciidoc/spring-cloud-config.adoc
@@ -983,6 +983,7 @@ keystore properties are `encrypt.keyStore.\*` with `*` equal to
 |`encrypt.keyStore.location`|Contains a `Resource` location
 |`encrypt.keyStore.password`|Holds the password that unlocks the keystore
 |`encrypt.keyStore.alias`|Identifies which key in the store to use
+|`encrypt.keyStore.type`|The type of KeyStore to create.  Defaults to `jks`.
 |===
 
 The encryption is done with the public key, and a private key is

--- a/pom.xml
+++ b/pom.xml
@@ -11,7 +11,7 @@
 	<parent>
 		<groupId>org.springframework.cloud</groupId>
 		<artifactId>spring-cloud-build</artifactId>
-		<version>2.0.5.BUILD-SNAPSHOT</version>
+		<version>2.0.6.BUILD-SNAPSHOT</version>
 		<relativePath/> <!-- lookup parent from repository -->
 	</parent>
 	<scm>
@@ -22,7 +22,7 @@
 	</scm>
     <properties>
         <bintray.package>config</bintray.package>
-		<spring-cloud-commons.version>2.0.3.BUILD-SNAPSHOT</spring-cloud-commons.version>
+		<spring-cloud-commons.version>2.0.4.BUILD-SNAPSHOT</spring-cloud-commons.version>
     </properties>
 	<modules>
 		<module>spring-cloud-config-dependencies</module>

--- a/spring-cloud-config-dependencies/pom.xml
+++ b/spring-cloud-config-dependencies/pom.xml
@@ -5,7 +5,7 @@
 	<parent>
 		<artifactId>spring-cloud-dependencies-parent</artifactId>
 		<groupId>org.springframework.cloud</groupId>
-		<version>2.0.5.BUILD-SNAPSHOT</version>
+		<version>2.0.6.BUILD-SNAPSHOT</version>
 		<relativePath/>
 	</parent>
 	<artifactId>spring-cloud-config-dependencies</artifactId>

--- a/spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/EncryptionAutoConfiguration.java
+++ b/spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/EncryptionAutoConfiguration.java
@@ -97,7 +97,7 @@ public class EncryptionAutoConfiguration {
 			KeyStore keyStore = this.key.getKeyStore();
 			KeyStoreTextEncryptorLocator locator = new KeyStoreTextEncryptorLocator(
 					new KeyStoreKeyFactory(keyStore.getLocation(),
-							keyStore.getPassword().toCharArray(), "jks"),
+							keyStore.getPassword().toCharArray(), key.getKeyStore().getType()),
 					keyStore.getSecret(), keyStore.getAlias());
 			RsaAlgorithm algorithm = this.rsaProperties.getAlgorithm();
 			locator.setRsaAlgorithm(algorithm);

--- a/spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/EncryptionAutoConfiguration.java
+++ b/spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/EncryptionAutoConfiguration.java
@@ -97,7 +97,7 @@ public class EncryptionAutoConfiguration {
 			KeyStore keyStore = this.key.getKeyStore();
 			KeyStoreTextEncryptorLocator locator = new KeyStoreTextEncryptorLocator(
 					new KeyStoreKeyFactory(keyStore.getLocation(),
-							keyStore.getPassword().toCharArray()),
+							keyStore.getPassword().toCharArray(), "jks"),
 					keyStore.getSecret(), keyStore.getAlias());
 			RsaAlgorithm algorithm = this.rsaProperties.getAlgorithm();
 			locator.setRsaAlgorithm(algorithm);

